src/include/OpenImageIO/paramlist.h

```diff
@@ -326,6 +327,9 @@ class OIIO_UTIL_API ParamValue {
                       Copy _copy                = Copy(true),
                       FromUstring _from_ustring = FromUstring(false)) noexcept;
     void clear_value() noexcept;
+
+    /// declare a friend heapsize definition
+    template<typename T> friend inline size_t heapsize(const T&);
```

Comment:
Does it not work to declare only the specialized one as a friend here? Like

    template<> friend size_t pvt::heapsize(const ParamValue&);

Reply:
I think when I've tried it didn't, but I can give it another go , just in case !